[Android] Back button doesn't go back a page but quits webview

On Android when navigating through websites I want to use the back button to go back a page. This is currently not possible, the webview will be closed. Is this intentional?

I use the example from the repository and can reproduce this behavior with the fullscreen and widget webview. I digged in the source code and it should be able to set the back button to go back in the webviews history first by setting a `setOnKeyListener` on the webview ([see here](https://stackoverflow.com/a/17652491)).

How is the behaviour on iOS? If I recall correctly, you swipe from left to right to go back a page in Safari. Is this currently possible?

For the Android part I can create a pull request to change the back button behaviour. Should it be configurable or always go back until we reach the end of the history before exiting the webview?

Thanks for your quick response, I can try to fix it. Can you give me a hint how to compile flutter_webview? I'm new to flutter + dart and when I try `flutter run` or `flutter build apk` inside the `examples` folder, I get the following error message:

```
Initializing gradle...                                0,8s
Resolving dependencies...
* Error running Gradle:
Exit code 1 from: C:\Code\flutter_webview_plugin\example\android\gradlew.bat app:properties:

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Code\flutter_webview_plugin\example\android\build.gradle' line: 20

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not resolve all dependencies for configuration ':app:_debugApk'.
      > A problem occurred configuring project ':flutter_webview_plugin'.
         > Failed to notify project evaluation listener.
            > com.android.build.gradle.tasks.factory.AndroidJavaCompile.setDependencyCacheDir(Ljava/io/File;)V

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s

Please review your Gradle project setup in the android/ folder.
```

I couldn't figue out if I have to compile the plugin itself before, but a `flutter build apk` in the root directory doesn't work either, because the flutter library isn't included as a dependency in the android project.

--- a/android/src/main/java/com/flutter_webview_plugin/WebviewManager.java
+++ b/android/src/main/java/com/flutter_webview_plugin/WebviewManager.java
@@ -3,6 +3,7 @@ package com.flutter_webview_plugin;
 import android.annotation.TargetApi;
 import android.app.Activity;
 import android.os.Build;
+import android.view.KeyEvent;
 import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.CookieManager;
@@ -26,6 +27,25 @@ class WebviewManager {
         this.webView = new WebView(activity);
         WebViewClient webViewClient = new BrowserClient();
         webView.setWebViewClient(webViewClient);
+
+        webView.setOnKeyListener(new View.OnKeyListener() {
+            @Override
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+
+                if (event.getAction() == KeyEvent.ACTION_DOWN) {
+                    switch (keyCode) {
+                        case KeyEvent.KEYCODE_BACK:
+                            if (webView.canGoBack()) {
+                                webView.goBack();
+                                return true;
+                            }
+                            break;
+                    }
+                }
+
+                return false;
+            }
+        });
     }
 
     private void clearCookies() {

